tests/mfx_dispatch/linux/mfx_dispatch_test_hooks.cpp

```diff
@@ -19,14 +19,59 @@
 // SOFTWARE.
 
 #include "mfx_dispatch_test_hooks.h"
+#include "mfx_dispatch_test_main.h"
+
+#include <gtest/gtest.h>
+#include <mfxcommon.h>
 #include <iostream>
+#include <vector>
+#include <string>
+#include <algorithm>
+
 // dlopen hooks implementation:
 
 void* TEST_DLOPEN_HOOKS::AlwaysNull(const char *filename, int flag)
 {
     return nullptr;
 }
 
+void* TEST_DLOPEN_HOOKS::AlwaysNullNameCheck(const char *filename, int flag, HookInternalParams par)
```

Comment:
This looks like different expectations mixed together for unknown reasons. IMO, it should be different cases, like
`CheckImplAuto`, `CheckImplHW`, `CheckImplSW`